snowfakery/data_generator_runtime_object_model.py

```diff
@@ -128,6 +128,10 @@ def name(self) -> str:
             name += " (self.nickname)"
         return name
 
+    @property
+    def id(self):
+        return id(self)
```

Comment:
Keep in mind that using id() can be problematic if these objects ever need to survive serialization or get swapped out for another object.

Reply:
I did consider that but the disambiguation is only needed for the lifetime of a recipe execution.